lib/libxdp/xsk.c

```diff
@@ -388,12 +388,10 @@ static int xsk_init_xsk_struct(struct xsk_socket *xsk, int ifindex)
 	return 0;
 }
 
-#define XDP_FLAGS_SUPPORTED (XDP_FLAGS_SKB_MODE | XDP_FLAGS_DRV_MODE | XDP_FLAGS_HW_MODE)
-
 static enum xdp_attach_mode xsk_convert_xdp_flags(__u32 xdp_flags)
 {
-	if (xdp_flags & ~XDP_FLAGS_SUPPORTED)
-		pr_warn("XDP flag not supported by libxdp.\n");
+	if (xdp_flags & ~XDP_FLAGS_MASK)
+		pr_warn("XDP flag: 0x%x not supported by libxdp.\n", xdp_flags);
```

Comment:
If we're printing the unsupported flags, shouldn't that be the masked value?

Reply:
The `xdp_flags` is usually from invoker, the printing here means to notify invoker the total flags are not fully supported.  
Whether it should be masked value depends on the perspective of the message we're printing.  
For me I think the whole value is better. :grin:

Reply:
Right, but then we should change the message so it's clearer. Something like: "XDP flags 0x%x contains flag values not supported by libxdp" or something like that...